master/internal/trial.go

```diff
@@ -312,7 +312,7 @@ func (t *trial) Receive(ctx *actor.Context) error {
 	case trialWatchPreemption:
 		// Size 2; at most 2 messages can be sent and we don't want to block or lose them.
 		w := make(chan bool, 2)
-		if t.PendingGracefulTermination {
+		if t.PendingGracefulTermination || t.experimentState != model.ActiveState {
```

Comment:
can you explain a bit why do you need this line?

Reply:
previously if you started listening to preemption signals for a paused trial it would say, no you dont have to preempt but wait and I'll tell you when. a trial paused trial (or anything but active) should always just tell the caller to exit.

Reply:
is this conflict with our plan to change the states of the experiments/trials?

Reply:
> is this conflict with our plan to change the states of the experiments/trials 

I have no plan to do this, but I did have a plan to add substates under active.

> Why isn't it handled in the termination of the trial?

because the caller may not be listening yet when the trial terminates.